src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -627,14 +632,48 @@ findPayer = runMaybeT $ do
     mkApp i r as = App (TVar r i) (map (liftTerm . fromPactValue) as) i
 
     runCap i capRef as input = do
-      ar <- evalCap i CapCallStack False $ mkApp i capRef as
+      msgBody <- enrichedMsgBody cmd
+      ar <- local (setEnvMsgBody msgBody) $
+        evalCap i CapCallStack False $ mkApp i capRef as
+
       case ar of
         NewlyAcquired -> do
           r <- input
           popCapStack (const (return ()))
           return r
         _ -> evalError' i "Internal error, GAS_PAYER already acquired"
 
+enrichedMsgBody :: Command (Payload PublicMeta ParsedCode) -> Eval e Value
+enrichedMsgBody cmd = case (_pPayload $ _cmdPayload cmd) of
+  Exec (ExecMsg (ParsedCode _ exps) userData) -> return $
+    object [ "tx-type" A..= ( "exec" :: Text)
+           , "exec-code" A..= map renderCompactText exps --TODO should we do both?
```

Comment:
I'm going to say just this and not `exec-exprs`.

`exec-exprs` is very well done so I regret ditching it, the justification is however that it's actually too low-level, because `Exp` is so low-level ~, so doing something as simple as matching `(coin.transfer "foo" "bar" 10.0)` will mean checking~

~1) `"list"` type 
2) first element is `(` separator 
3) 2nd is `"coin"` literal
4) then `.` separator
5) then `"transfer"` literal~ EDIT I since saw how it's done in the tests, it's not this bad

Note that this is brittle, in that the expressions `(use coin) (transfer "foo" "bar" 10.0)` is the same but will fail the above. 

Going with just code means the gas-payer will have to provide explicit instructions about code requirements, ie "you must directly reference calls to `coin.transfer` to qualify for gas payment". So let's start with just that.

EDIT it's not as bad as above, but forcing gas-payer to just compare strings is cheaper and less powerful which I think is actually good. If somebody submits a module to a gas payer the Exp JSON will be huge.

Reply:
yeah just having the code simplifies things. I fix it in the latest commit